src/main/java/com/lmax/disruptor/BatchEventProcessor.java

```diff
@@ -49,19 +51,44 @@
      * @param dataProvider    to which events are published.
      * @param sequenceBarrier on which it is waiting.
      * @param eventHandler    is the delegate to which events are dispatched.
+     * @param maxBatchSize    limits number of events processed in a batch before updating the sequence.
      */
     public BatchEventProcessor(
         final DataProvider<T> dataProvider,
         final SequenceBarrier sequenceBarrier,
-        final EventHandler<? super T> eventHandler)
+        final EventHandler<? super T> eventHandler,
+        final int maxBatchSize
+    )
     {
         this.dataProvider = dataProvider;
         this.sequenceBarrier = sequenceBarrier;
         this.eventHandler = eventHandler;
+        if (maxBatchSize < 1)
+        {
+            throw new IllegalArgumentException("maxBatchSize must be greater than 0");
+        }
+        this.batchLimitOffset = maxBatchSize - 1;
 
         eventHandler.setSequenceCallback(sequence);
     }
 
+    /**
+     * Construct a {@link EventProcessor} that will automatically track the progress by updating its sequence when
+     * the {@link EventHandler#onEvent(Object, long, boolean)} method returns.
+     *
+     * @param dataProvider    to which events are published.
+     * @param sequenceBarrier on which it is waiting.
+     * @param eventHandler    is the delegate to which events are dispatched.
+     */
+    public BatchEventProcessor(
+            final DataProvider<T> dataProvider,
+            final SequenceBarrier sequenceBarrier,
+            final EventHandler<? super T> eventHandler
+    )
+    {
+        this(dataProvider, sequenceBarrier, eventHandler, Integer.MAX_VALUE);
```

Comment:
Can we replace this `Integer.MAX_VALUE` with a constant? `DEFAULT_MAX_BATCH_SIZE` maybe?